### PR TITLE
OSX renaming

### PR DESCRIPTION
**Pull request by dominikl**
Took over @sbesson 's PR #58 so I can continue working on it without the need for @sbesson to merge every change into his PR.

**_Test**_ : Check status of https://ci.openmicroscopy.org/job/OMERO-DEV-merge-homebrew/

---

Description from the original PR:

This PR is an attempt at unifying the OSX installation convention with the changes proposed in #56.

Main changes are:
- the top-level is renamed as osx instead of homebrew to match with other top-level folders
- the installation scripts to be executed (and exposed in the doc) are renamed as step*.sh
- a top-level install_osx script is created executes all the steps

Following up this PR, the CI job will need updating to execute something along the lines of

`cd osx &&  bash cleanup_homebrew.sh && bash install_homebrew.sh`

The scripts can also be copied as part of the autogen job like Linux walkthrough and directly embedded into the documentation for OSX.

Open questions:
- script termination: with or without .sh ?
- Bio-Formats C++ formula: is this still necessary/appropriate in this repo?
- naming convention: is install_osx okay (trying to match the Linux scheme)

### CODE REVIEW

File: osx/step04_test.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+brew install wget
+sleep 30
+wget --keep-session-cookies --save-cookies cookies.txt http://localhost:$HTTPPORT/webclient/login/ -O csrf_index.html
+csrf=$(cat csrf_index.html | grep "name=\'csrfmiddlewaretoken\'"  | sed "s/.* value=\'\(.*\)\'.*/\1/")
+post_data="username=root&password=$ROOT_PASSWORD&csrfmiddlewaretoken=$csrf&server=1&noredirect=1"
+resp=$(wget --keep-session-cookies --load-cookies cookies.txt --post-data $post_data http://localhost:$HTTPPORT/webclient/login/)
+echo "$resp"
+
+# Stop OMERO.web
+nginx -c $(bin/brew --prefix omero52)/etc/nginx.conf -s stop
```
Comment (sbesson) [inline osx/step04_test.sh]: One more `bin/brew` to remove - see https://ci.openmicroscopy.org/job/OMERO-DEV-merge-homebrew/ICE=3.5,OSX=snapper/49/console

**Comment by dominikl** [comment]
Looking at the original PR the open questions seem to be resolved, except _Bio-Formats C++ formula: is this still necessary/appropriate in this repo?_ Does this need further discussion @sbesson  ?

**Comment by sbesson** [comment]
I would leave the discussion for a separate cleanup PR.

**Comment by pwalczysko** [comment]
Leaving out the Open Questions, this looks good, the job is green.

**Comment by jburel** [comment]
@dominikl: Do you have a card for the clean-up? It could go under 5.2.2 Deployment

**Comment by jburel** [comment]
Thanks all, it can be sorted in another PR. Merging so @hflynn can move forward on docs PR front

**Pull request merged**